internal/mithril-persistence/src/sqlite/connection_extensions.rs

```diff
@@ -13,6 +13,20 @@ pub trait ConnectionExtensions {
         sql: Q,
         params: &[Value],
     ) -> StdResult<T>;
+
+    /// Fetch entities from the database using the given query.
+    fn fetch<Q: Query>(&self, query: Q) -> StdResult<EntityCursor<Q::Entity>>;
+
+    /// Fetch the first entity from the database returned using the given query.
+    fn fetch_one<Q: Query>(&self, query: Q) -> StdResult<Option<Q::Entity>> {
```

Comment:
Agree for `fetch_first`.
For `fetch_all` it would introduce confusion with a query that is parametrized to fetch all row in a table (ie: `connection.fetch_all(GetTransaction::all())`), instead I just simplified from `fetch_and_collect` to just `fetch_collect`.